### ISSUE TITLE
Pale Moon Greasemonkey fork - long term intent?

### ISSUE THREAD

**Issue by VLM1** [janekptacijarabaci/greasemonkey#2]
Greetings janekptacijarabaci, and thank you for creating an up-to-date fork of Greasemonkey for Pale Moon users. I do have a question about your intentions. On your page [https://github.com/janekptacijarabaci/greasemonkey/releases/tag/3.9Fork](url), you write:

> Will be updated sporadically - or maybe never...
No, I don't want to fork the project (Greasemonkey).

Although I would, naturally, prefer to use your forked version of current-generation Greasemonkey, I wonder whether I (all PM users) should take this path for the long term. What is your intention, if you don't mind my asking?

**Comment by janekptacijarabaci** [comment]
Hello.

The following events occur (soon)...

(order may be reversed)

__1) Releasing a new version of Pale Moon (v27)__

See ["Pale Moon Forum" - "Ad: The Firefox addon Greasemonkey 3.8 (the original version)"]
(https://forum.palemoon.org/viewtopic.php?f=56&t=12557)

Maybe (Greasemonkey code is constantly changing) it will not be needed (the fork).

But then there is another event:

__2) Greasemonkey move from XPCOM/XUL to WebExtensions (if so be it - but it is very likely)__

See also ["Pale Moon Forum" - "Greasemonkey recent versions"](https://forum.palemoon.org/viewtopic.php?f=16&t=10293&p=71318#p71318)

Maybe in the future will be added (e.g., Moonchild wrote: "WebExtensions is not currently on the radar because we have much more important things to tackle first" - see https://forum.palemoon.org/viewtopic.php?t=12486#p88362)...

But... There are too many incompatible changes for Pale Moon.

Maybe:
* __Bugs__: I will try to fix __*serious*__ bugs and make a new release.
* __Enhancements__: If it's a simple code change, I can do it.

Or at least temporarily?!

As for me (if nobody takes him - see https://addons.palemoon.org/extensions/all-extensions/), I'm still undecided... It depends on many factors.

We'll see...

**Comment by VLM1** [comment]
janekptacijarabaci, thank you for your kind response. 

For better or for worse, I don't believe the PM dev team intends to add WebExtensions during any foreseeable time frame, if for no other reason than that the resources needed to support such an effort are apparently not available. I can understand that, of course. 

If you are willing, as you say, to fix serious bugs, then at least those of us using PM pre-27 can continue to use the extension. OTOH, given the uncertainty involved, although I have been considering learning how to create some UserScripts for PM, I think I won't and will focus elsewhere instead. 

Best regards,

**Comment by janekptacijarabaci** [comment]
When it released a new version of Pale Moon... Fixing bugs for the older version...
I don't know if I'll do it.

It belong mainly to: "Greasemonkey move from XPCOM/XUL to WebExtensions".

**Comment by VLM1** [comment]
Yes, I understand, very clear. Thank you so much for this conversation.

**Comment by janekptacijarabaci** [comment]
You're welcome :-)

**Comment by janekptacijarabaci** [comment]
See https://forum.palemoon.org/viewtopic.php?f=46&t=9501&start=40#p98409

**Comment by filbo** [comment]
Is greasemonkey#2456 a request to import this entire fork into mainline GM, or does it leave some things behind?

**Comment by filbo** [comment]
A: GM 2456 ~~ 8 files, +53 -43 lines
[this fork](https://github.com/janekptacijarabaci/greasemonkey/compare/3.9...3.9.2Fork) ~~ 26 files, +857 -167 lines

... so what exactly are you proposing in 2456?

**Comment by janekptacijarabaci** [comment]
This fork is (mainly) for Pale Moon 26.5.0-. It's currently enough for Pale Moon 27+.

**Comment by Lootyhoof** [comment]
You know, one way or another I'd really like to get a version of Greasemonkey onto the Pale Moon Add-Ons Site. Given the fact that time is lacking for the official Greasemonkey (plus you're the most active contributor there lately, given the number of [pull requests](https://github.com/greasemonkey/greasemonkey/pulls)) I'd like to get this version live on the site, if you're willing. If you're not that's fine also, but if you are do you have a more private form of communication available to relay information to (it seems forum PMs are disabled for your account)?

**Comment by janekptacijarabaci** [comment]
@Lootyhoof 

I'm sorry.

Unlike version 3.9.2-, version 3.9.3+ is an experimental version...
And moreover... It isn't a full fledged fork at this point. Why?
\- It’s simply a question of time and responsibility.

I haven't decided what will happen. For now I leave it as it is for later evaluation...

But of course... Can you paste a link to your webpage. But I don't know if that's enough for you now...

> it seems forum PMs are disabled for your account

I don't know anything about it.

**Comment by Lootyhoof** [comment]
Understandable. I can certainly link to this repository if you'd prefer, it's mostly just a matter of making the public at large aware that Greasemonkey does work on Pale Moon in one form or another.

It also seems that I do have the option of sending a PM now...odd.

**Comment by janekptacijarabaci** [comment]
> I can certainly link to this repository if you'd prefer

Yes, thank you. And maybe it will stay that way :-)

**Comment by TPS** [comment]
@janekptacijarabaci There's been quite a bit of activity on the PM branch since last ß, but no release recently.  Could that be rolled up into 1, please? :bow:

**Comment by janekptacijarabaci** [comment]
@TPS 
See https://forum.palemoon.org/viewtopic.php?f=46&t=16840#p123290
(it will be released as soon as possible)

**Comment by TPS** [comment]
@janekptacijarabaci Thanks much for all your work, & I look forward to the mentioned impending release. 👏

**Comment by yetiisyou** [comment]
Good time! Please fix work with https://github.com/SavageCore/yadg-pth-userscript  https://rawgit.com/SavageCore/yadg-pth-userscript/master/pth_yadg.user.js

**Comment by janekptacijarabaci** [comment]
@yetiisyou 

Please:
- Create a new issue
- Give as much detail as possible (i.e. any information from Browser Console are very helpful)
